src/v/coproc/script_context.cc

```diff
@@ -29,32 +32,24 @@
 
 namespace coproc {
 
-struct batch_info {
-    model::offset last;
-    std::size_t total_size_bytes;
-    model::record_batch_reader rbr;
-};
-
-std::optional<batch_info>
-extract_batch_info(model::record_batch_reader::data_t data) {
-    if (data.empty()) {
-        /// TODO(rob) Its possible to recieve an empty batch when a
-        /// batch type filter is enabled on the reader which for copro
-        return std::nullopt;
+class high_offset_tracker {
+public:
+    struct info {
```

Comment:
`info` is generally a bad name if public. but ok since it's inside.

Reply:
does `info` need to be public?

Reply:
yes it does :)

Reply:
isn't this a `batch_metdata_info`, plain `info` seems too generic.

Reply:
changing the name